pkgs/tools/networking/vpnc-scripts/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib, stdenv, fetchgit
+, makeWrapper
+, nettools, gawk, openresolv, coreutils, gnugrep
+}:
+
+let
+  os = if stdenv.isDarwin then "Darwin" else "Linux";
+
+in stdenv.mkDerivation {
+  pname = "vpnc-scripts";
+  version = "unstable-2020-02-21";
+  src = fetchgit {
+    url = "git://git.infradead.org/users/dwmw2/vpnc-scripts.git";
+    rev = "c0122e891f7e033f35f047dad963702199d5cb9e";
+    sha256 = "11b1ls012mb704jphqxjmqrfbbhkdjb64j2q4k8wb5jmja8jnd14";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp vpnc-script $out/bin
+
+    substituteInPlace $out/bin/vpnc-script \
+      --replace "which" "type -P" \
+      --replace "/sbin/resolvconf" "${openresolv}/bin/resolvconf"
+
+    wrapProgram $out/bin/vpnc-script \
+      --set OS ${os} \
+      --prefix PATH : "${lib.makeBinPath [ nettools gawk openresolv coreutils gnugrep ]}"
+  '';
+
```

Comment:
```suggestion
preFixup = ''
    substituteInPlace $out/bin/vpnc-script \
      --replace "which" "type -P"
  '' + lib.optionalString stdenv.isLinux ''
    \ --replace "/sbin/resolvconf" "${openresolv}/bin/resolvconf"
  '' + ''
    wrapProgram $out/bin/vpnc-script \
      --set OS ${os} \
      --prefix PATH : "${lib.makeBinPath vpncBinPath}"
  '';
```

Reply:
I would move the whole wrapping of the executable to the `preFixup` phase and make use of the `lib.optionalString` function to take care of the conditional os setup.

Just out of curiosity: Why is `--set OS ${os}` needed?

Reply:
Hm, I thought it was because the script couldn't determine the value of `$OS` on its own.  But re-reading, it unconditionally does:

```
OS="`uname -s`"
```

... so it seems useless.  I'll take it out and make sure it still works.